core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -766,6 +785,29 @@ class RequestMetrics(name: String) extends KafkaMetricsGroup {
     }
   }
 
+  class Histogram(val metricNamePrefix: String, val metricNamePostfix: String, val binBoundaries: Array[Int]) {
```

Comment:
Int -> Long considering the time is measured in ms, including the type of the key in the TreeMap etc

Reply:
For latency yes, but just in case this Histogram class is used for something else, e.g. size of requests.

Reply:
If we need that big number, which probably indicates we should use a different unit. For request size, we set the boundary with the unit of Mb currently, and the biggest configured boundary is 100 Mb. Even if we change it Kb in future, Int should also be good enough.

Reply:
You are right that Int can satisfy the current needs.
But I feel this class needs to be a bit more generic and future-proof, just as how the yammer metrics library stores the values using Long.